File: U1/TI1.3 RESTFULL_JAVA_SINBDD_GR01/02 CLIESC/ConUni_CliEsc_RestFull_Java_GR01/ConUni_CliEsc_GR01/ec/edu/monster/models/ConversionResultModel.cs
namespace ConUni_CliEsc_GR01.ec.edu.monster.models;

/// <summary>
/// Modelo de respuesta de la API de conversión
/// </summary>
public class ConversionResultModel
{
    /// <summary>
    /// Indica si la conversión fue exitosa
    /// </summary>
    public bool Exitoso { get; set; }

    /// <summary>
    /// Resultado de la conversión (presente si Exitoso = true)
    /// </summary>
    public UnidadConversionModel? Resultado { get; set; }

    /// <summary>
    /// Información del error (presente si Exitoso = false)
    /// </summary>
    public ConversionErrorModel? Error { get; set; }
}

File: U1/TI1.3 RESTFULL_JAVA_SINBDD_GR01/02 CLICON/ConUni_Rest_Cliente_Java_GR01/src/main/java/ec/edu/monster/models/ConversionResultModel.java
package ec.edu.monster.models;

/**
 * Modelo para recibir resultados de conversión del servidor REST
 * Representa el JSON que se recibe como respuesta
 * 
 * Ejemplo JSON exitoso:
 * {
 *   "exitoso": true,
 *   "resultado": {
 *     "valorOriginal": 100.0,
 *     "valorConvertidoExacto": 0.0621371192,
 *     "valorConvertidoRedondeado": 0.062,
 *     "unidadOrigen": "Metro",
 *     "unidadDestino": "Milla",
 *     "factorConversion": 0.000621371192
 *   },
 *   "error": null
 * }
 * 
 * @author jeffe
 */
public class ConversionResultModel {
    
    private boolean exitoso;
    private UnidadConversion resultado;
    private ConversionError error;
    
    // Constructor vacío
    public ConversionResultModel() {
    }
    
    // Getters y Setters
    public boolean isExitoso() {
        return exitoso;
    }
    
    public void setExitoso(boolean exitoso) {
        this.exitoso = exitoso;
    }
    
    public UnidadConversion getResultado() {
        return resultado;
    }
    
    public void setResultado(UnidadConversion resultado) {
        this.resultado = resultado;
    }
    
    public ConversionError getError() {
        return error;
    }
    
    public void setError(ConversionError error) {
        this.error = error;
    }
    
    @Override
    public String toString() {
        return "ConversionResultModel{" +
               "exitoso=" + exitoso +
               ", resultado=" + resultado +
               ", error=" + error +
               '}';
    }
}